### ISSUE TITLE
TrailingWhitespace false positive on KDoc line wraps

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5713]
## Steps to Reproduce
```kotlin
/**
 * Something.  
 * Something else.
 */
fun f() {}
```

## Observed Behavior
The two spaces on the first textual comment line above (not really visible on GitHub, select the code block to see) is highlighted and reported as a finding by `TrailingWhitespace`.

## Expected Behavior
No report, valid KDoc.

## Context
In Java the above comment would be written as:
```java
/**
 * Something.<br>
 * Something else.
 */
```
in Kotlin, this is not supported, because KDoc uses Markdown. In markdown a forced line-wrap is achieved by adding two trailing spaces at the end of the line. Without this the rendered output would be
```
Something. Something else.
```
while the author explicitly wants:
```
Something.
Something else.
```

Sadly Kotlin hasn't implemented what they [claim ("regular markdown")](https://kotlinlang.org/docs/kotlin-doc.html#inline-markup) yet:
https://youtrack.jetbrains.com/issue/KTIJ-6702/KDoc-Dokka-allow-for-newlines-line-breaks-inside-paragraphs#focus=Comments-27-6792193.0-0

## Your Environment

* Version of detekt used: 1.22.0
* Version of Gradle used (if applicable): 7.6
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): N/A, full repro above

**Comment by cortinico** [comment]
> a forced line-wrap is achieved by adding two trailing spaces at the end of the line

TIL

This is up for grab if someone is looking for a relatively easy contribution 👍

**Comment by cortinico** [comment]
> Sadly Kotlin hasn't implemented what they [claim ("regular markdown")](https://kotlinlang.org/docs/kotlin-doc.html#inline-markup) yet:
[youtrack.jetbrains.com/issue/KTIJ-6702/KDoc-Dokka-allow-for-newlines-line-breaks-inside-paragraphs#focus=Comments-27-6792193.0-0](https://youtrack.jetbrains.com/issue/KTIJ-6702/KDoc-Dokka-allow-for-newlines-line-breaks-inside-paragraphs#focus=Comments-27-6792193.0-0)

Also I don't think this is a reason to block this @TWiStErRob right?  We could just extend our logic to support it?

**Comment by TWiStErRob** [comment]
We could support it, but if KDoc/Dokka doesn't render it, then it's really just trailing whitespace for no reason 😞. Let's see what JetBrains says first. We could even close this as "blocked" for now.
